Application/components/FriendsColumn/RemoveFriendModal.tsx

```diff
@@ -0,0 +1,148 @@
+
+  const friendOptions = friendsList.map((friend: Friend) => ({
+    value: friend.id,
+    label: friend.username,
+  }));
+
+  const handleOpenModalClick = () => {
+    if (friendsList.length > 0) {
+      setOpened(true);
+    } else {
+      showNotification({
+        title: 'No Friends to Remove',
+        message: 'You currently have no friends to remove.😢',
+        color: 'red',
+      });
+    }
+  };
+
+  const handleRemoveFriendClick = async (): Promise<void> => {
+    if (selectedFriend.length === 0) {
+      setErrorMessages({ friends: "Please select a friend to remove." });
+      return;
+    }
+
+    try {
+      const response = await fetch('/api/Remove-friend', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          userID: user?.id,
+          friendIDs: selectedFriend,
+        }),
+      });
+
+      if (response.ok) {
+        // Iterate over selectedFriend array and show a notification for each
+        selectedFriend.forEach(friendId => {
+          const friendName = friendOptions.find(option => option.value === friendId)?.label || 'Unknown friend';
+          showNotification({
+            title: 'Friend Removed',
+            message: `✅ ${friendName} was removed successfully.`,
+            color: 'green',
+          });
+        });
+
+        setSelectedFriend([]);
+        setErrorMessages({ friends: '' });
+        setLastFetched(Date.now());
+        setOpened(false);
+      } else {
+        const errorText: string = await response.text();
+        showNotification({
+          title: 'Error',
+          message: errorText || 'Failed to remove friend.',
+          color: 'red',
+        });
+      }
+    } catch (error: any) {
+      console.error('Error removing friend:', error);
+      showNotification({
+        title: 'Error',
+        message: error.toString() || 'An error occurred while removing the friend.',
+        color: 'red',
+      });
+    }
+  };
+
+  return (
+    <>
+      <Button
+        onClick={handleOpenModalClick}
+        variant="subtle"
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+          gap: '10px',
+          backgroundImage: 'linear-gradient(to right, #E14D7D, #FF9D0A)',
+          color: 'white',
+          border: 'none',
+          borderRadius: '5px',
+          padding: '8px 16px',
+        }}
+      >
+        <IconUserMinus size={16} style={{ color: 'white', marginRight: 5 }} />
+        Remove a friend
+      </Button>
+
+      <Modal
+        centered
+        opened={opened}
+        onClose={() => setOpened(false)}
+        title={
+          <Stack gap="0">
+            <Text variant="gradient" fw={500} size="xl" component="span">
+              Remove a Friend
+            </Text>
+            <Text color={theme.colors.dark[1]}>Select a friend to remove</Text>
```

Comment:
Great to see you accessing the `theme.colors.dark` theme! This is configured within the RootLayout, and this is how you access certain color scheme tones.